Fixed several bugs (including #4)

zivnadel opened the pull request:
I have fixed several bugs in this Pull Request.

Firstly, I addressed issue #4. I fixed it by doing something similar to what @alex-laycalvert suggested there.
I changed `if(focusedItem)` to `if(focusedItem != -1)` in `generateStateWithNewFolder` and `generateStateWithNewFolder` (located in `FileSystemHelpers.ts`). In my opinion, this change is more accurate because `item.index` will either be `-1` or a string. Using `if(focusedItem >= 0)` would work but is less clear.

Secondly, I addressed some bugs related to PR #22 regarding focusing a file. Since in that PR I changed the structure of the file system item (making the index and the object key the path of the file) it broke some stuff:
- It broke dragging and dropping files. I fixed it by adapting the new changes in the helper functions (inside `FileSystemHelpers.ts`)
- In `addFile` and `addFolder` functions (located in `FileSystem.tsx`), the check for the existence of an item was broken because the key is no longer the name but the path. I implemented a new function in `FileSystemHelpers.tsx` to make this check, now that the name is a property (`data`) and not the object key.

**_Important note: I have not yet adapted the `handleRenameItem` function (located in `FileSystem.tsx`) to the changes in PR #22. Therefore, renaming will likely be broken when it is implemented. I will make sure to fix it as soon as possible._**

I believe these changes have resolved all known issues. However, if any further bugs are discovered, I will be sure to fix them as soon as possible.

zivnadel committed:
Fixed a bug where a user can't create files if there are no files and more bugs

yonatanmgr reviewed (changes requested):
Hey Ziv, that's amazing! #4 is indeed practically solved now, thank you very much!
I am having some issues when creating new folders (does not raise the modal and pops a system error instead, and proceeds to create a new folder anyway).
In addition, creating a new file while a file with that name exists anywhere in the file system opens the "A new file already exists" modal - I don't think that's intentional (as we're checking the file path and not the name since #22 if I'm not mistaken).
What do you think?

zivnadel commented:
I’ll Look into it.

zivnadel committed:
Fixed some regression bugs

zivnadel commented:
I have fixed the two bugs that you mentioned. The first one was caused by a typo I made in the `generateStateWithNewFolder` function, where I mistakenly replaced `newFolderName` with `newFileName`.

The second bug was caused by the `checkIfItemNameIsFolder` function, which checked for the existence of files with a certain name throughout the system. I have now modified it to check for items by the provided name under a passed node (path) - in our use case, the focused item. Although I believe that this solution works well for now, we should consider passing the parent node when a **folder is opened**, not just when it is focused. If this indicator has already been implemented, please let me know.

zivnadel committed:
Fixed bugs in the filesystem

zivnadel committed:
Clean up

zivnadel commented:
@yonatanmgr I fixed the last bugs, and made some changes. The main change I did is that `generateStateWithNewFile`, `generateStateWithNewFolder` and `itemExistsInParent` (which is the new version of `checkIfItemNameIsFolder`, more about that below) now relate to the `selectedFolder` and not `focusedItem`. Other changes I made:
- Recreated `checkIfItemNameIsFolder` as `itemExistsInParent`. It is pretty much doing the same, but now correct and more concise. Reference the comments in the function for more information.
- Shortened `generateStateWithNewFile` and `generateStateWithNewFolder`. The first `if` check in those functions was redundant, since it was looking for the parent's path, which we don't need to look for now, as it is the key (or `index`).

yonatanmgr reviewed (approved):
Just reviewed it - amazing. 
Thank you!
